libs/designer/src/lib/ui/panel/panelTabs/parametersTab/index.tsx

```diff
@@ -274,19 +274,42 @@ const ParameterSection = ({
     editorId: string,
     labelId: string,
     tokenPickerMode?: TokenPickerMode,
+    editorType?: string,
     isCodeEditor?: boolean,
     closeTokenPicker?: () => void,
     tokenPickerClicked?: (b: boolean) => void,
     tokenClickedCallback?: (token: ValueSegment) => void
   ): JSX.Element => {
-    const codeEditorFilteredTokens = tokenGroup.filter((group) => {
-      return group.id !== 'workflowparameters' && group.id !== 'variables';
-    });
+    const parameterType =
+      editorType ??
+      (nodeInputs.parameterGroups[group.id].parameters.find((param) => param.id === parameterId) ?? {})?.type ??
+      constants.SWAGGER.TYPE.ANY;
+    const supportedTypes: string[] = parameterType
+      ? getPropertyValue(constants.TOKENS, parameterType) || [constants.SWAGGER.TYPE.ANY, parameterType]
+      : null;
+
+    const filteredTokenGroup = tokenGroup.map((group) => ({
+      ...group,
+      tokens: group.tokens.filter((token: OutputToken) => {
+        if (isCodeEditor) {
+          return !(
+            token.outputInfo.functionName === constants.FUNCTION_NAME.VARIABLES ||
+            token.outputInfo.functionName === constants.FUNCTION_NAME.PARAMETERS ||
```

Comment:
Please use token.outputInfo.tokenType to check for variables or parameters, we have different type of tokens already stamped in the object, do not use functionName directly, it is supposed to be only used during serialization if needed